### DISCUSSION TITLE
Odd json serialization issue

### DISCUSSION THREAD

**Discussion by bbartling** [fastapi/fastapi#11364] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/bacnet/write", response_model=BaseResponse)
async def bacnet_write_property(request_body: WritePropertyRequest, request: Request, username: str = Depends(get_current_username)):
    if _debug:
        # Access the raw request body if needed
        raw_body = await request.body()
        _log.debug(f"request_body Raw request body: {raw_body}")
        _log.debug(f"request_body Parsed BACnet POST data: {request_body.json()}")
        _log.debug(f"request_body Parsed BACnet POST data: {request_body.dict()}")

    request = await request.json()

    if _debug:
        _log.debug(f"request Parsed BACnet POST data: {request}")
    
    # Extracting the request data
    device_instance = request['device_instance']
    object_identifier = request['object_identifier']
    property_identifier = request['property_identifier']
    value = request['value']
    priority = request['priority']
```
```


### Description

I have POST route on my app with a pydantic model where when the JSON POST data is validate it seems like my key value pair is returning as Null or None.



### Operating System

Linux

### Operating System Details

Ubuntu flavor of embedded Linux Armbian ran on rasbian hardware.

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.10.12

### Additional Context

This is my fast API POST route:

```
@app.post("/bacnet/write", response_model=BaseResponse)
async def bacnet_write_property(request_body: WritePropertyRequest, request: Request, username: str = Depends(get_current_username)):
    if _debug:
        # Access the raw request body if needed
        raw_body = await request.body()
        _log.debug(f"request_body Raw request body: {raw_body}")
        _log.debug(f"request_body Parsed BACnet POST data: {request_body.json()}")
        _log.debug(f"request_body Parsed BACnet POST data: {request_body.dict()}")

    request = await request.json()

    if _debug:
        _log.debug(f"request Parsed BACnet POST data: {request}")
    
    # Extracting the request data
    device_instance = request['device_instance']
    object_identifier = request['object_identifier']
    property_identifier = request['property_identifier']
    value = request['value']
    priority = request['priority']
```

Where this data when validating with my model `request_body: WritePropertyRequest` this key value is `'object_identifier': None` or `null` like there is an issue with serializing `analog-value,301` but it works just fine if using `request.json()`. This is more debug log prints. 

```
DEBUG:__main__:request_body Raw request body: b'{\n  "device_instance": 201201,\n  "object_identifier": "analog-value,301",\n  "property_identifier": "present-value",\n  "value": 134,\n  "priority": 10\n}'
DEBUG:__main__:request_body Parsed BACnet POST data: {"device_instance":201201,"object_identifier":null,"property_identifier":"present-value","value":134.0,"priority":10}
DEBUG:__main__:request_body Parsed BACnet POST data: {'device_instance': 201201, 'object_identifier': None, 'property_identifier': 'present-value', 'value': 134, 'priority': 10}
DEBUG:__main__:request Parsed BACnet POST data: {'device_instance': 201201, 'object_identifier': 'analog-value,301', 'property_identifier': 'present-value', 'value': 134, 'priority': 10}
DEBUG:__main__:Request Device Instance: 201201
DEBUG:__main__:Request Object Identifier: analog-value,301
DEBUG:__main__:Request Property Identifier: present-value
DEBUG:__main__:Request Value: 134
DEBUG:__main__:Request Priority: 10
```

It feels a bit hacky but if i use `request_body: WritePropertyRequest` to validate my pydantic model everything works just fine and then use `request: Request` for processing the data and returning results...

Would anyone have any advice? I am using the Swagger UI to test all this.

**Comment by bbartling** [comment, marked as answer]
sorry for the formatting...

**Reply by YuriiMotov** [reply]
You can edit it
